### ISSUE TITLE
Examples don't build

### ISSUE THREAD

**Issue by linse** [janestreet/incr_dom#5]
Hi friends, I tried your library and followed the example build explanation at https://github.com/janestreet/incr_dom/blob/master/example/README.org.
Unfortunately, the build step did not work for me, and fails with
```
File "src/incr.ml", line 3, characters 14-30:
Error: Unbound module Incremental
```
I `opam install`ed incremental, so I don't understand why this happens. Can you help?

**Comment by ghost** [comment]
This is odd indeed. Could you provide the full build log?

**Comment by sgpthomas** [comment]
I have the same problem.

```
# # # Workspace root: /home/samthomas/Development/incr_dom
# # Auto-detected concurrency: 4
# Dune context:
#  ((name default)
#   (kind default)
#   (profile dev)
#   (merlin true)
#   (for_host ())
#   (build_dir (In_build_dir default))
#   (toplevel_path ((External /home/samthomas/.opam/4.06.1/lib/toplevel)))
#   (ocaml_bin (External /home/samthomas/.opam/4.06.1/bin))
#   (ocaml (External /home/samthomas/.opam/4.06.1/bin/ocaml))
#   (ocamlc (External /home/samthomas/.opam/4.06.1/bin/ocamlc.opt))
#   (ocamlopt ((External /home/samthomas/.opam/4.06.1/bin/ocamlopt.opt)))
#   (ocamldep (External /home/samthomas/.opam/4.06.1/bin/ocamldep.opt))
#   (ocamlmklib (External /home/samthomas/.opam/4.06.1/bin/ocamlmklib.opt))
#   (env
#    ((DUNE_CONFIGURATOR /home/samthomas/.opam/4.06.1/bin/ocamlc.opt)
#     (INSIDE_DUNE 1)
#     (OCAMLPATH
#      /home/samthomas/Development/incr_dom/_build/install/default/lib)))
#   (findlib_path ((External /home/samthomas/.opam/4.06.1/lib)))
#   (arch_sixtyfour true)
#   (natdynlink_supported true)
#   (opam_vars ())
#   (ocaml_config
#    ((version 4.06.1)
#     (standard_library_default /home/samthomas/.opam/4.06.1/lib/ocaml)
#     (standard_library /home/samthomas/.opam/4.06.1/lib/ocaml)
#     (standard_runtime /home/samthomas/.opam/4.06.1/bin/ocamlrun)
#     (ccomp_type cc)
#     (c_compiler gcc)
#     (ocamlc_cflags (-O2 -fno-strict-aliasing -fwrapv -fPIC))
#     (ocamlopt_cflags (-O2 -fno-strict-aliasing -fwrapv))
#     (bytecomp_c_compiler
#      (gcc
#       -O2
#       -fno-strict-aliasing
#       -fwrapv
#       -fPIC
#       -D_FILE_OFFSET_BITS=64
#       -D_REENTRANT))
#     (bytecomp_c_libraries (-lm -ldl -lcurses -lpthread))
#     (native_c_compiler
#      (gcc
#       -O2
#       -fno-strict-aliasing
#       -fwrapv
#       -D_FILE_OFFSET_BITS=64
#       -D_REENTRANT))
#     (native_c_libraries (-lm -ldl))
#     (cc_profile (-pg))
#     (architecture amd64)
#     (model default)
#     (int_size 63)
#     (word_size 64)
#     (system linux)
#     (asm (as))
#     (asm_cfi_supported true)
#     (with_frame_pointers false)
#     (ext_exe "")
#     (ext_obj .o)
#     (ext_asm .s)
#     (ext_lib .a)
#     (ext_dll .so)
#     (os_type Unix)
#     (default_executable_name a.out)
#     (systhread_supported true)
#     (host x86_64-unknown-linux-gnu)
#     (target x86_64-unknown-linux-gnu)
#     (profiling true)
#     (flambda false)
#     (spacetime false)
#     (safe_string false)
#     (exec_magic_number Caml1999X011)
#     (cmi_magic_number Caml1999I022)
#     (cmo_magic_number Caml1999O022)
#     (cma_magic_number Caml1999A022)
#     (cmx_magic_number Caml1999Y022)
#     (cmxa_magic_number Caml1999Z022)
#     (ast_impl_magic_number Caml1999M022)
#     (ast_intf_magic_number Caml1999N022)
#     (cmxs_magic_number Caml1999D022)
#     (cmt_magic_number Caml1999T022)
#     (natdynlink_supported true)))
#   (which
#    ((ocaml ((External /home/samthomas/.opam/4.06.1/bin/ocaml)))
#     (ocamlc ((External /home/samthomas/.opam/4.06.1/bin/ocamlc.opt))))))
# Actual targets:
# - _build/default/example/event_propagation/main.bc.js
# - _build/default/example/event_propagation/index.html
Running[2]: (cd _build/default && /home/samthomas/.opam/4.06.1/bin/ocamlc.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -bin-annot -I src/.incr_dom.objs -I /home/samthomas/.opam/4.06.1/lib/async_js -I /home/samthomas/.opam/4.06.1/lib/async_kernel -I /home/samthomas/.opam/4.06.1/lib/async_rpc_kernel -I /home/samthomas/.opam/4.06.1/lib/base -I /home/samthomas/.opam/4.06.1/lib/base/caml -I /home/samthomas/.opam/4.06.1/lib/base/md5 -I /home/samthomas/.opam/4.06.1/lib/base/shadow_stdlib -I /home/samthomas/.opam/4.06.1/lib/bin_prot -I /home/samthomas/.opam/4.06.1/lib/bin_prot/shape -I /home/samthomas/.opam/4.06.1/lib/bytes -I /home/samthomas/.opam/4.06.1/lib/core -I /home/samthomas/.opam/4.06.1/lib/core_kernel -I /home/samthomas/.opam/4.06.1/lib/core_kernel/base_for_tests -I /home/samthomas/.opam/4.06.1/lib/fieldslib -I /home/samthomas/.opam/4.06.1/lib/incr_map -I /home/samthomas/.opam/4.06.1/lib/incr_select -I /home/samthomas/.opam/4.06.1/lib/incremental -I /home/samthomas/.opam/4.06.1/lib/incremental_kernel -I /home/samthomas/.opam/4.06.1/lib/jane-street-headers -I /home/samthomas/.opam/4.06.1/lib/js_of_ocaml -I /home/samthomas/.opam/4.06.1/lib/ocaml/threads -I /home/samthomas/.opam/4.06.1/lib/parsexp -I /home/samthomas/.opam/4.06.1/lib/ppx_assert/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_bench/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_compare/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/collector -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/common -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/config -I /home/samthomas/.opam/4.06.1/lib/ppx_hash/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_inline_test/config -I /home/samthomas/.opam/4.06.1/lib/ppx_inline_test/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_sexp_conv/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/protocol_version_header -I /home/samthomas/.opam/4.06.1/lib/re -I /home/samthomas/.opam/4.06.1/lib/re/posix -I /home/samthomas/.opam/4.06.1/lib/seq -I /home/samthomas/.opam/4.06.1/lib/sexplib -I /home/samthomas/.opam/4.06.1/lib/sexplib/unix -I /home/samthomas/.opam/4.06.1/lib/sexplib0 -I /home/samthomas/.opam/4.06.1/lib/spawn -I /home/samthomas/.opam/4.06.1/lib/splittable_random -I /home/samthomas/.opam/4.06.1/lib/stdio -I /home/samthomas/.opam/4.06.1/lib/stringext -I /home/samthomas/.opam/4.06.1/lib/typerep -I /home/samthomas/.opam/4.06.1/lib/tyxml -I /home/samthomas/.opam/4.06.1/lib/uchar -I /home/samthomas/.opam/4.06.1/lib/uri -I /home/samthomas/.opam/4.06.1/lib/uutf -I /home/samthomas/.opam/4.06.1/lib/variantslib -I /home/samthomas/.opam/4.06.1/lib/virtual_dom -no-alias-deps -opaque -open Incr_dom__ -o src/.incr_dom.objs/incr_dom__Incr.cmo -c -impl src/incr.pp.ml)

Command [2] exited with code 2:
$ (cd _build/default && /home/samthomas/.opam/4.06.1/bin/ocamlc.opt -w @a-4-29-40-41-42-44-45-48-58-59-60-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -bin-annot -I src/.incr_dom.objs -I /home/samthomas/.opam/4.06.1/lib/async_js -I /home/samthomas/.opam/4.06.1/lib/async_kernel -I /home/samthomas/.opam/4.06.1/lib/async_rpc_kernel -I /home/samthomas/.opam/4.06.1/lib/base -I /home/samthomas/.opam/4.06.1/lib/base/caml -I /home/samthomas/.opam/4.06.1/lib/base/md5 -I /home/samthomas/.opam/4.06.1/lib/base/shadow_stdlib -I /home/samthomas/.opam/4.06.1/lib/bin_prot -I /home/samthomas/.opam/4.06.1/lib/bin_prot/shape -I /home/samthomas/.opam/4.06.1/lib/bytes -I /home/samthomas/.opam/4.06.1/lib/core -I /home/samthomas/.opam/4.06.1/lib/core_kernel -I /home/samthomas/.opam/4.06.1/lib/core_kernel/base_for_tests -I /home/samthomas/.opam/4.06.1/lib/fieldslib -I /home/samthomas/.opam/4.06.1/lib/incr_map -I /home/samthomas/.opam/4.06.1/lib/incr_select -I /home/samthomas/.opam/4.06.1/lib/incremental -I /home/samthomas/.opam/4.06.1/lib/incremental_kernel -I /home/samthomas/.opam/4.06.1/lib/jane-street-headers -I /home/samthomas/.opam/4.06.1/lib/js_of_ocaml -I /home/samthomas/.opam/4.06.1/lib/ocaml/threads -I /home/samthomas/.opam/4.06.1/lib/parsexp -I /home/samthomas/.opam/4.06.1/lib/ppx_assert/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_bench/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_compare/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/collector -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/common -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/config -I /home/samthomas/.opam/4.06.1/lib/ppx_hash/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_inline_test/config -I /home/samthomas/.opam/4.06.1/lib/ppx_inline_test/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_sexp_conv/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/protocol_version_header -I /home/samthomas/.opam/4.06.1/lib/re -I /home/samthomas/.opam/4.06.1/lib/re/posix -I /home/samthomas/.opam/4.06.1/lib/seq -I /home/samthomas/.opam/4.06.1/lib/sexplib -I /home/samthomas/.opam/4.06.1/lib/sexplib/unix -I /home/samthomas/.opam/4.06.1/lib/sexplib0 -I /home/samthomas/.opam/4.06.1/lib/spawn -I /home/samthomas/.opam/4.06.1/lib/splittable_random -I /home/samthomas/.opam/4.06.1/lib/stdio -I /home/samthomas/.opam/4.06.1/lib/stringext -I /home/samthomas/.opam/4.06.1/lib/typerep -I /home/samthomas/.opam/4.06.1/lib/tyxml -I /home/samthomas/.opam/4.06.1/lib/uchar -I /home/samthomas/.opam/4.06.1/lib/uri -I /home/samthomas/.opam/4.06.1/lib/uutf -I /home/samthomas/.opam/4.06.1/lib/variantslib -I /home/samthomas/.opam/4.06.1/lib/virtual_dom -no-alias-deps -opaque -open Incr_dom__ -o src/.incr_dom.objs/incr_dom__Incr.cmo -c -impl src/incr.pp.ml)
File "src/incr.ml", line 2, characters 14-30:
Error: Unbound module Incremental
```

**Comment by hhugo** [comment]
To build the examples, you'll have to use the very latest version of our libraries 

```
opam repo add janestreet-bleeding https://ocaml.janestreet.com/opam-repository
opam update
opam upgrade
```

**Comment by sgpthomas** [comment]
Incremental fails to build.

```
The following actions will be performed:
  ∗  install incremental v0.11.0

=-=- Gathering sources =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[incremental] Archive in cache

=-=- Processing actions -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
[ERROR] The compilation of incremental failed at "jbuilder build -p incremental -j 4".

#=== ERROR while installing incremental.v0.11.0 ===============================#
# opam-version 1.2.2
# os           linux
# command      jbuilder build -p incremental -j 4
# path         /home/samthomas/.opam/4.06.1/build/incremental.v0.11.0
# compiler     4.06.1
# exit-code    1
# env-file     /home/samthomas/.opam/4.06.1/build/incremental.v0.11.0/incremental-16068-1872ec.env
# stdout-file  /home/samthomas/.opam/4.06.1/build/incremental.v0.11.0/incremental-16068-1872ec.out
# stderr-file  /home/samthomas/.opam/4.06.1/build/incremental.v0.11.0/incremental-16068-1872ec.err
### stderr ###
# [...]
# File "src/incremental.ml", line 8, characters 24-38:
# Error: Unbound module Weak_hashtbl
#     ocamlopt src/.incremental_lib.objs/incremental_lib__Incremental.{cmx,o} (exit 2)
# (cd _build/default && /home/samthomas/.opam/4.06.1/bin/ocamlopt.opt -w -40 -safe-string -g -I src/.incremental_lib.objs -I /home/samthomas/.opam/4.06.1/lib/base -I /home/samthomas/.opam/4.06.1/lib/base/caml -I /home/samthomas/.opam/4.06.1/lib/base/md5 -I /home/samthomas/.opam/4.06.1/lib/base/shadow_stdlib -I /home/samthomas/.opam/4.06.1/lib/base_quickcheck -I /home/samthomas/.opam/4.06.1/lib/bin_prot -I /home/samthomas/.opam/4.06.1/lib/bin_prot/shape -I /home/samthomas/.opam/4.06.1/lib/core -I /home/samthomas/.opam/4.06.1/lib/core_kernel -I /home/samthomas/.opam/4.06.1/lib/core_kernel/base_for_tests -I /home/samthomas/.opam/4.06.1/lib/fieldslib -I /home/samthomas/.opam/4.06.1/lib/incremental_kernel -I /home/samthomas/.opam/4.06.1/lib/jane-street-headers -I /home/samthomas/.opam/4.06.1/lib/ocaml/threads -I /home/samthomas/.opam/4.06.1/lib/parsexp -I /home/samthomas/.opam/4.06.1/lib/ppx_assert/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_bench/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_compare/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_enumerate/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/collector -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/common -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/config -I /home/samthomas/.opam/4.06.1/lib/ppx_hash/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_inline_test/config -I /home/samthomas/.opam/4.06.1/lib/ppx_inline_test/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_sexp_conv/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/sexplib -I /home/samthomas/.opam/4.06.1/lib/sexplib/unix -I /home/samthomas/.opam/4.06.1/lib/sexplib0 -I /home/samthomas/.opam/4.06.1/lib/spawn -I /home/samthomas/.opam/4.06.1/lib/splittable_random -I /home/samthomas/.opam/4.06.1/lib/stdio -I /home/samthomas/.opam/4.06.1/lib/typerep -I /home/samthomas/.opam/4.06.1/lib/variantslib -intf-suffix .ml -no-alias-deps -open Incremental_lib__ -o src/.incremental_lib.objs/incremental_lib__Incremental.cmx -c -impl src/incremental.pp.ml)
# File "src/incremental.ml", line 8, characters 24-38:
# Error: Unbound module Weak_hashtbl
#       ocamlc src/.incremental_lib.objs/incremental_lib__Incremental_unit_tests.{cmo,cmt} (exit 2)
# (cd _build/default && /home/samthomas/.opam/4.06.1/bin/ocamlc.opt -w -40 -safe-string -g -bin-annot -I src/.incremental_lib.objs -I /home/samthomas/.opam/4.06.1/lib/base -I /home/samthomas/.opam/4.06.1/lib/base/caml -I /home/samthomas/.opam/4.06.1/lib/base/md5 -I /home/samthomas/.opam/4.06.1/lib/base/shadow_stdlib -I /home/samthomas/.opam/4.06.1/lib/base_quickcheck -I /home/samthomas/.opam/4.06.1/lib/bin_prot -I /home/samthomas/.opam/4.06.1/lib/bin_prot/shape -I /home/samthomas/.opam/4.06.1/lib/core -I /home/samthomas/.opam/4.06.1/lib/core_kernel -I /home/samthomas/.opam/4.06.1/lib/core_kernel/base_for_tests -I /home/samthomas/.opam/4.06.1/lib/fieldslib -I /home/samthomas/.opam/4.06.1/lib/incremental_kernel -I /home/samthomas/.opam/4.06.1/lib/jane-street-headers -I /home/samthomas/.opam/4.06.1/lib/ocaml/threads -I /home/samthomas/.opam/4.06.1/lib/parsexp -I /home/samthomas/.opam/4.06.1/lib/ppx_assert/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_bench/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_compare/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_enumerate/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/collector -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/common -I /home/samthomas/.opam/4.06.1/lib/ppx_expect/config -I /home/samthomas/.opam/4.06.1/lib/ppx_hash/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_inline_test/config -I /home/samthomas/.opam/4.06.1/lib/ppx_inline_test/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/ppx_sexp_conv/runtime-lib -I /home/samthomas/.opam/4.06.1/lib/sexplib -I /home/samthomas/.opam/4.06.1/lib/sexplib/unix -I /home/samthomas/.opam/4.06.1/lib/sexplib0 -I /home/samthomas/.opam/4.06.1/lib/spawn -I /home/samthomas/.opam/4.06.1/lib/splittable_random -I /home/samthomas/.opam/4.06.1/lib/stdio -I /home/samthomas/.opam/4.06.1/lib/typerep -I /home/samthomas/.opam/4.06.1/lib/variantslib -intf-suffix .ml -no-alias-deps -open Incremental_lib__ -o src/.incremental_lib.objs/incremental_lib__Incremental_unit_tests.cmo -c -impl src/incremental_unit_tests.pp.ml)
# File "src/incremental_unit_tests.ml", line 1681, characters 29-36:
# Error: Unbound value Int.gen



=-=- Error report -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
The following actions failed
  ∗  install incremental v0.11.0
No changes have been performed
```

I'm on switch 4.06.1.
![image](https://user-images.githubusercontent.com/3752122/45464773-0b1a4280-b6e0-11e8-83ef-fc2a54bca76b.png)

**Comment by TyOverby** [comment]
My virtualdom fails to build with the following error

```
#=== ERROR while installing virtual_dom.v0.11.120.01+170 ======================#
# opam-version 1.2.2
# os           darwin
# command      jbuilder build -p virtual_dom -j 4
# path         /Users/tyoverby/.opam/system/build/virtual_dom.v0.11.120.01+170
# compiler     system (4.07.0)
# exit-code    1
# env-file     /Users/tyoverby/.opam/system/build/virtual_dom.v0.11.120.01+170/virtual_dom-54500-e7d71c.env
# stdout-file  /Users/tyoverby/.opam/system/build/virtual_dom.v0.11.120.01+170/virtual_dom-54500-e7d71c.out
# stderr-file  /Users/tyoverby/.opam/system/build/virtual_dom.v0.11.120.01+170/virtual_dom-54500-e7d71c.err
### stderr ###
      ocamlc src/.virtual_dom.objs/virtual_dom__Attr_intf.{cmi,cmo,cmt} (exit 2)
# (cd _build/default && /usr/local/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.virtual_dom.objs -I src/.virtual_dom.objs/.private -I /Users/tyoverby/.opam/system/lib/base -I /Users/tyoverby/.opam/system/lib/base/caml -I /Users/tyoverby/.opam/system/lib/base/md5 -I /Users/tyoverby/.opam/system/lib/base/shadow_stdlib -I /Users/tyoverby/.opam/system/lib/base_quickcheck -I /Users/tyoverby/.opam/system/lib/bin_prot -I /Users/tyoverby/.opam/system/lib/bin_prot/shape -I /Users/tyoverby/.opam/system/lib/bytes -I /Users/tyoverby/.opam/system/lib/core_kernel -I /Users/tyoverby/.opam/system/lib/core_kernel/base_for_tests -I /Users/tyoverby/.opam/system/lib/fieldslib -I /Users/tyoverby/.opam/system/lib/jane-street-headers -I /Users/tyoverby/.opam/system/lib/js_of_ocaml -I /Users/tyoverby/.opam/system/lib/parsexp -I /Users/tyoverby/.opam/system/lib/ppx_assert/runtime-lib -I /Users/tyoverby/.opam/system/lib/ppx_bench/runtime-lib -I /Users/tyoverby/.opam/system/lib/ppx_compare/runtime-lib -I /Users/tyoverby/.opam/system/lib/ppx_enumerate/runtime-lib -I /Users/tyoverby/.opam/system/lib/ppx_expect/collector -I /Users/tyoverby/.opam/system/lib/ppx_expect/common -I /Users/tyoverby/.opam/system/lib/ppx_expect/config -I /Users/tyoverby/.opam/system/lib/ppx_hash/runtime-lib -I /Users/tyoverby/.opam/system/lib/ppx_inline_test/config -I /Users/tyoverby/.opam/system/lib/ppx_inline_test/runtime-lib -I /Users/tyoverby/.opam/system/lib/ppx_sexp_conv/runtime-lib -I /Users/tyoverby/.opam/system/lib/re -I /Users/tyoverby/.opam/system/lib/seq -I /Users/tyoverby/.opam/system/lib/sexplib -I /Users/tyoverby/.opam/system/lib/sexplib0 -I /Users/tyoverby/.opam/system/lib/splittable_random -I /Users/tyoverby/.opam/system/lib/stdio -I /Users/tyoverby/.opam/system/lib/typerep -I /Users/tyoverby/.opam/system/lib/tyxml -I /Users/tyoverby/.opam/system/lib/uchar -I /Users/tyoverby/.opam/system/lib/uutf -I /Users/tyoverby/.opam/system/lib/variantslib -I css/src/.css.objs -no-alias-deps -open Virtual_dom__ -o src/.virtual_dom.objs/virtual_dom__Attr_intf.cmo -c -impl src/attr_intf.pp.ml)
# File "src/attr_intf.ml", line 1:
# Error: Wrong file naming: /Users/tyoverby/.opam/system/lib/js_of_ocaml/css.cmi
# contains the compiled interface for 
#
```

**Comment by TyOverby** [comment]
Interestingly, I don't have a `...../css.cmi` file on my computer.  There is a `...../cSS.cmi` file though.  This repros on osx where it has a case-insensitive filesystem, but osx doesn't turn files lowercase in listings.  

Given the last command (an invocation of ocamlc.opt) I expect that this is a compiler bug, though it could be a build system bug as well (I don't see dune passing in a reference to css.cmi though).

**Comment by hhugo** [comment]
@

**Comment by hhugo** [comment]
@sgpthomas, are you using the bleeding edge opam-repository described above ? can you run `opam remote -v` ?
did you call `opam update` and `opam upgrade` ?

**Comment by hhugo** [comment]
@TyOverby, your issue is already tracked there https://github.com/janestreet/virtual_dom/issues/3

**Comment by sgpthomas** [comment]
![image](https://user-images.githubusercontent.com/3752122/45758772-64a8d280-bbf4-11e8-8224-d7267a8fd95d.png)

Yes, I'm on the bleeding edge repo. I've called `opam update && opam upgrade` many times.

**Comment by TyOverby** [comment]
@hhugo: ah, so it is.  For whatever reason I thought virtual_dom lived in this repo.

**Comment by hhugo** [comment]
@sgpthomas, I was able to reproduce your issue. The problem is that some of the janestreet packages (in that case, `Incremental`) are no compatible with ocaml < 4.07 anymore. It means that even using the bleeding edge janestreet repo, you won't be able to install the latest version of incremental (and incr_dom) on ocaml 4.06. Can you upgrade to 4.07 ?

**Comment by rauanmayemir** [comment]
Upgrading to `4.07` helps, but the build is still blocked by `virtual_dom` issue. (under macOS because of case insensitive filesystem)

**Comment by sgpthomas** [comment]
upgrading to 4.07.0 fixes it for me
